experiments/conductor/cmd/runner/crd_generator_commands.go

```diff
@@ -0,0 +1,320 @@
+
+The generate.sh script is located at apis/<SERVICE>/v1alpha1/generate.sh.
+
+Add the parameter <TICK> --resource <CRD_KIND>:<PROTO_RESOURCE><TICK> to the <TICK>go run . generate-types --api-version <CRD_GROUP>/<CRD_VERSION>  <TICK> command.
+
+At the end of the script, ensure the following lines are present:
+
+cd ${REPO_ROOT}
+dev/tasks/generate-crds
+
+go run -mod=readonly golang.org/x/tools/cmd/goimports@latest -w pkg/controller/direct/<SERVICE>/
+`
+
+// NOT USED
+func generateCRDFromScripts(opts *RunnerOptions, branch Branch) {
+	close := setLoggingWriter(opts, branch)
+	defer close()
+	workDir := filepath.Join(opts.branchRepoDir)
+
+	var out strings.Builder
+	checkoutBranch(branch, workDir, &out)
+
+	// Create the apis/<service>/<version> directory
+	serviceDir := filepath.Join(workDir, "apis", branch.Group, "v1alpha1")
+	if err := os.MkdirAll(serviceDir, 0755); err != nil {
+		log.Fatal(err)
+	}
+
+	// Create or update generate.sh
+	scriptPath := filepath.Join(serviceDir, "generate.sh")
+	// Check if generate.sh already exists.
+	if _, err := os.Stat(scriptPath); errors.Is(err, os.ErrNotExist) {
+		// File doesn't exist, use template approach
+		log.Printf("Creating new generate.sh at %s", scriptPath)
+
+		// Replace template markers with actual values and write to file
+		writeTemplateToFile(branch, scriptPath, GENERATOR_SCRIPT_TEMPLATE)
+	} else {
+		// File exists, use codebot to update it
+		log.Printf("Updating existing generate.sh at %s", scriptPath)
+
+		// Delete then write the prompt file.
+		promptPath := filepath.Join(workDir, "mockgcp", "crdgen_prompt.txt")
+		writeTemplateToFile(branch, promptPath, UPDATE_GENERATE_SCRIPT_PROMPT)
+
+		cfg := CommandConfig{
+			Name:    "CODEBOT GENERATE",
+			Cmd:     "codebot",
+			Args:    []string{"--ui-type=prompt", "--prompt=mockgcp/crdgen_prompt.txt"},
+			WorkDir: workDir,
+		}
+		if err := executeCommand(cfg, &out); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	// Add and commit changes
+	scriptRelativePath := fmt.Sprintf("apis/%s/v1alpha1/generate.sh", branch.Group)
+	gitAdd(workDir, &out, scriptRelativePath)
+	gitCommit(workDir, &out, fmt.Sprintf("add/update crd generation script for %s", branch.Group))
+
+	// Run the generator script
+	cfg := CommandConfig{
+		Name:    "Generator script",
+		Cmd:     scriptPath,
+		WorkDir: workDir,
+	}
+	if err := executeCommand(cfg, &out); err != nil {
+		log.Fatal(err)
+	}
+
+	// Add and commit generated files
+	gitAdd(workDir, &out,
+		fmt.Sprintf("apis/%s/v1alpha1/", branch.Group),
+		fmt.Sprintf("pkg/controller/direct/%s/", branch.Group),
+		"config/crds/resources/")
+	gitCommit(workDir, &out, fmt.Sprintf("autogenerated types and CRDs using %s", scriptRelativePath))
+}
+
+func generateTypesAndMapper(opts *RunnerOptions, branch Branch) {
+	close := setLoggingWriter(opts, branch)
+	defer close()
+	workDir := filepath.Join(opts.branchRepoDir)
```

Comment:
filepath.Join is about getting the slashes correct. If your not concatenating directories/files, you don't need to call it.

Reply:
done.